src/java.base/share/classes/sun/security/provider/certpath/OCSP.java

```diff
@@ -224,18 +225,27 @@ static OCSPResponse check(List<CertId> certIds, URI responderURI,
         OCSPRequest request = new OCSPRequest(certIds, extensions);
         byte[] bytes = request.encodeBytes();
 
-        InputStream in = null;
-        OutputStream out = null;
-        byte[] response = null;
-
-        try {
-            URL url = responderURI.toURL();
-            if (debug != null) {
-                debug.println("connecting to OCSP service at: " + url);
-            }
-
-            Event.report(Event.ReporterCategory.CRLCHECK, "event.ocsp.check", url.toString());
-            HttpURLConnection con = (HttpURLConnection)url.openConnection();
+        if (debug != null) {
+            debug.println("connecting to OCSP service at: " + responderURI);
+        }
+        Event.report(Event.ReporterCategory.CRLCHECK, "event.ocsp.check",
+                responderURI.toString());
+
+        URL url;
+        HttpURLConnection con;
+        String encodedGetReq = responderURI.toString() + "/" +
+                URLEncoder.encode(Base64.getMimeEncoder(0, new byte[0]).
```

Comment:
Force of habit.  I'm so used to using getMimeEncoder for other structures that need line breaks so I just stuck with it.  From the docs it looks like the basic one might work just fine.  The encoder returned by Base64.getUrlEncoder() definitely doesn't give me the encoding I want, which is why I went with the MIME encoder and then pushed it through URLEncoder.encode() to get the proper HTTP path string value.  I'll try it with the Basic base64 encoder, since that will simplify the code a bit.